pkg/genericapiserver/config.go

```diff
@@ -0,0 +1,467 @@
+		MinRequestTimeout: time.Duration(c.MinRequestTimeout) * time.Second,
+
+		MasterCount:          c.MasterCount,
+		ExternalAddress:      c.ExternalHost,
+		ClusterIP:            c.PublicAddress,
+		PublicReadWritePort:  c.ReadWritePort,
+		ServiceReadWriteIP:   c.ServiceReadWriteIP,
+		ServiceReadWritePort: c.ServiceReadWritePort,
+		ExtraServicePorts:    c.ExtraServicePorts,
+		ExtraEndpointPorts:   c.ExtraEndpointPorts,
+
+		KubernetesServiceNodePort: c.KubernetesServiceNodePort,
+		apiGroupsForDiscovery:     map[string]unversioned.APIGroup{},
+
+		enableOpenAPISupport:   c.EnableOpenAPISupport,
+		openAPIInfo:            c.OpenAPIInfo,
+		openAPIDefaultResponse: c.OpenAPIDefaultResponse,
+	}
+
+	if c.RestfulContainer != nil {
+		s.mux = c.RestfulContainer.ServeMux
+		s.HandlerContainer = c.RestfulContainer
+	} else {
+		mux := http.NewServeMux()
+		s.mux = mux
+		s.HandlerContainer = NewHandlerContainer(mux, c.Serializer)
+	}
+	// Use CurlyRouter to be able to use regular expressions in paths. Regular expressions are required in paths for example for proxy (where the path is proxy/{kind}/{name}/{*})
+	s.HandlerContainer.Router(restful.CurlyRouter{})
+	s.MuxHelper = &apiserver.MuxHelper{Mux: s.mux, RegisteredPaths: []string{}}
+
+	if c.ProxyDialer != nil || c.ProxyTLSClientConfig != nil {
+		s.ProxyTransport = utilnet.SetTransportDefaults(&http.Transport{
+			Dial:            c.ProxyDialer,
+			TLSClientConfig: c.ProxyTLSClientConfig,
+		})
+	}
+
+	// Register root handler.
+	// We do not register this using restful Webservice since we do not want to surface this in api docs.
+	// Allow GenericAPIServer to be embedded in contexts which already have something registered at the root
+	if c.EnableIndex {
+		s.mux.HandleFunc("/", apiserver.IndexHandler(s.HandlerContainer, s.MuxHelper))
+	}
+
+	if c.EnableLogsSupport {
+		apiserver.InstallLogsSupport(s.MuxHelper, s.HandlerContainer)
+	}
+	if c.EnableUISupport {
+		ui.InstallSupport(s.MuxHelper, s.enableSwaggerSupport && s.enableSwaggerUI)
+	}
+
+	if c.EnableProfiling {
+		s.mux.HandleFunc("/debug/pprof/", pprof.Index)
+		s.mux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+		s.mux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	}
+
+	apiserver.InstallVersionHandler(s.MuxHelper, s.HandlerContainer)
+
+	handler := http.Handler(s.mux.(*http.ServeMux))
+
+	// TODO: handle CORS and auth using go-restful
+	// See github.com/emicklei/go-restful/blob/master/examples/restful-CORS-filter.go, and
+	// github.com/emicklei/go-restful/blob/master/examples/restful-basic-authentication.go
+
+	if len(c.CorsAllowedOriginList) > 0 {
+		allowedOriginRegexps, err := util.CompileRegexps(c.CorsAllowedOriginList)
+		if err != nil {
+			glog.Fatalf("Invalid CORS allowed origin, --cors-allowed-origins flag was set to %v - %v", strings.Join(c.CorsAllowedOriginList, ","), err)
+		}
+		handler = apiserver.CORS(handler, allowedOriginRegexps, nil, nil, "true")
+	}
+
+	s.InsecureHandler = handler
```

Comment:
why is insecureHandler written twice. Keep the temporary value local. This avoids any kind of hidden closures of the temporary value.